src/Stratis.Bitcoin/P2P/Protocol/Payloads/ProvenHeadersPayload.cs

```diff
@@ -29,6 +29,11 @@ public ProvenHeadersPayload(params ProvenBlockHeader[] headers)
             this.Headers.AddRange(headers);
         }
 
+        public ProvenHeadersPayload(IEnumerable<ProvenBlockHeader> headers)
```

Comment:
Why not just use the method above?

Reply:
to save an .array() on that linq code, do you want to remove it?

Reply:
yeah I prefer not to use linq in this components